File: Scripts/results/generated_code_results/RN/Social Networking/SocialSoftwareApp/ChatsScreen.js
import React, { useState } from 'react';

import { View, Text, Image, TextInput, TouchableOpacity, FlatList, StyleSheet } from 'react-native';

import { useNavigation } from '@react-navigation/native';



const ChatsScreen = () => {

  const [searchText, setSearchText] = useState('');

  const navigation = useNavigation();



  const stories = [

    { name: 'Your story', add: true },

    { name: 'Joshua', active: true },

    { name: 'Martin', active: true },

    { name: 'Karen', active: true },

    { name: 'Martha', active: true },

  ];



  const chatList = [

    { name: 'Martin Randolph', lastMessage: "You: What's man!", time: '9:40 AM' },

    { name: 'Andrew Parker', lastMessage: 'You: Ok, thanks!', time: '9:25 AM', read: true },

    { name: 'Karen Castillo', lastMessage: 'You: Ok, See you in To...', time: 'Fri', read: true },

    { name: 'Maisy Humphrey', lastMessage: 'Have a good day, Maisy!', time: 'Fri', read: true },

    { name: 'Joshua Lawrence', lastMessage: 'The business plan loo...', time: 'Thu', read: true },

  ];



  const renderStoryItem = ({ item }) => (

    <TouchableOpacity style={styles.storyItem}>

      <Image

        source={require('../assets/snack-icon.png')}

        style={[styles.storyImage, item.add && styles.addStoryImage]}

      />

      {item.active && <View style={styles.activeIndicator} />}

      <Text style={styles.storyName}>{item.name}</Text>

    </TouchableOpacity>

  );



  const renderChatItem = ({ item }) => (

    <TouchableOpacity

      style={styles.chatItem}

      onPress={() => navigation.navigate('Intro', { chatName: item.name })}

    >

      <Image source={require('../assets/snack-icon.png')} style={styles.chatImage} />

      <View style={styles.chatInfo}>

        <Text style={styles.chatName}>{item.name}</Text>

        <Text style={styles.chatLastMessage}>{item.lastMessage}</Text>

      </View>

      <Text style={styles.chatTime}>{item.time}</Text>

      {item.read && <View style={styles.readIndicator} />}

    </TouchableOpacity>

  );



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <Image source={require('../assets/snack-icon.png')} style={styles.profilePic} />

        <Text style={styles.title}>Chats</Text>

        <TouchableOpacity onPress={() => {/* Open camera */}}>

          <Image source={require('../assets/snack-icon.png')} style={styles.icon} />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('NewMessage')}>

          <Image source={require('../assets/snack-icon.png')} style={styles.icon} />

        </TouchableOpacity>

      </View>



      <TextInput

        style={styles.searchBar}

        placeholder="Search"

        value={searchText}

        onChangeText={setSearchText}

      />



      <FlatList

        horizontal

        data={stories}

        renderItem={renderStoryItem}

        keyExtractor={(item) => item.name}

        style={styles.storiesList}

      />



      <FlatList

        data={chatList}

        renderItem={renderChatItem}

        keyExtractor={(item) => item.name}

        style={styles.chatList}

      />



      <View style={styles.adContainer}>

        <Image source={require('../assets/snack-icon.png')} style={styles.adIcon} />

        <View style={styles.adInfo}>

          <Text style={styles.adTitle}>Pixsellz</Text>

          <Text style={styles.adMessage}>Make design process easier...</Text>

        </View>

        <TouchableOpacity>

          <Text style={styles.viewMore}>View More</Text>

        </TouchableOpacity>

      </View>



      <View style={styles.bottomNav}>

        <TouchableOpacity style={styles.navItem}>

          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />

          <Text>Chats</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem} onPress={() => navigation.navigate('People')}>

          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />

          <Text>People</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.navItem}>

          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />

          <Text>Discover</Text>

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#fff',

  },

  header: {

    flexDirection: 'row',

    alignItems: 'center',

    padding: 10,

  },

  profilePic: {

    width: 40,

    height: 40,

    borderRadius: 20,

  },

  title: {

    fontSize: 20,

    fontWeight: 'bold',

    marginLeft: 10,

  },

  icon: {

    width: 24,

    height: 24,

    marginLeft: 10,

  },

  searchBar: {

    backgroundColor: '#f0f0f0',

    padding: 10,

    margin: 10,

    borderRadius: 20,

  },

  storiesList: {

    paddingLeft: 10,

  },

  storyItem: {

    alignItems: 'center',

    marginRight: 15,

  },

  storyImage: {

    width: 60,

    height: 60,

    borderRadius: 30,

  },

  addStoryImage: {

    borderWidth: 2,

    borderColor: '#ccc',

  },

  activeIndicator: {

    position: 'absolute',

    bottom: 20,

    right: 0,

    width: 12,

    height: 12,

    borderRadius: 6,

    backgroundColor: 'green',

    borderWidth: 2,

    borderColor: 'white',

  },

  storyName: {

    marginTop: 5,

    fontSize: 12,

  },

  chatList: {

    flex: 1,

  },

  chatItem: {

    flexDirection: 'row',

    padding: 10,

    alignItems: 'center',

  },

  chatImage: {

    width: 50,

    height: 50,

    borderRadius: 25,

  },

  chatInfo: {

    flex: 1,

    marginLeft: 10,

  },

  chatName: {

    fontWeight: 'bold',

  },

  chatLastMessage: {

    color: '#777',

  },

  chatTime: {

    color: '#777',

    fontSize: 12,

  },

  readIndicator: {

    width: 10,

    height: 10,

    borderRadius: 5,

    backgroundColor: 'blue',

    marginLeft: 5,

  },

  adContainer: {

    flexDirection: 'row',

    alignItems: 'center',

    padding: 10,

    backgroundColor: '#f0f0f0',

  },

  adIcon: {

    width: 40,

    height: 40,

  },

  adInfo: {

    flex: 1,

    marginLeft: 10,

  },

  adTitle: {

    fontWeight: 'bold',

  },

  adMessage: {

    color: '#777',

  },

  viewMore: {

    color: 'blue',

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    padding: 10,

    borderTopWidth: 1,

    borderTopColor: '#ccc',

  },

  navItem: {

    alignItems: 'center',

  },

  navIcon: {

    width: 24,

    height: 24,

  },

});



export default ChatsScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Social Networking/SocialSoftwareApp/ChatsScreen.dart
import 'package:flutter/material.dart';
class ChatsPage extends StatefulWidget {
@override
_ChatsPageState createState() => _ChatsPageState();
}
class _ChatsPageState extends State<ChatsPage> {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Chats'),
leading: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
actions: [
IconButton(
icon: Icon(Icons.camera_alt),
onPressed: () {
// Open camera
},
),
IconButton(
icon: Icon(Icons.edit),
onPressed: () {
Navigator.pushNamed(context, 'NewMessage');
},
),
],
),
body: Column(
children: [
Padding(
padding: EdgeInsets.all(8.0),
child: TextField(
decoration: InputDecoration(
hintText: 'Search',
prefixIcon: Icon(Icons.search),
border: OutlineInputBorder(
borderRadius: BorderRadius.circular(20),
),
),
),
),
Container(
height: 100,
child: ListView(
scrollDirection: Axis.horizontal,
children: [
_buildStoryItem('Your story', isAdd: true),
_buildStoryItem('Joshua', isActive: true),
_buildStoryItem('Martin', isActive: true),
_buildStoryItem('Karen', isActive: true),
_buildStoryItem('Martha', isActive: true),
],
),
),
Expanded(
child: ListView(
children: [
_buildChatItem('Martin Randolph', "You: What's man!", '9:40 AM'),
_buildChatItem('Andrew Parker', 'You: Ok, thanks!', '9:25 AM', isRead: true),
_buildChatItem('Karen Castillo', 'You: Ok, See you in To...', 'Fri', isRead: true),
_buildChatItem('Maisy Humphrey', 'Have a good day, Maisy!', 'Fri', isRead: true),
_buildChatItem('Joshua Lawrence', 'The business plan loo...', 'Thu', isRead: true),
_buildAdItem(),
],
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 0,
items: [
BottomNavigationBarItem(icon: Icon(Icons.chat), label: 'Chats'),
BottomNavigationBarItem(icon: Icon(Icons.people), label: 'People'),
BottomNavigationBarItem(icon: Icon(Icons.explore), label: 'Discover'),
],
onTap: (index) {
if (index == 1) {
Navigator.pushNamed(context, 'People');
}
},
),
);
}
Widget _buildStoryItem(String name, {bool isAdd = false, bool isActive = false}) {
return Container(
width: 70,
margin: EdgeInsets.symmetric(horizontal: 4),
child: Column(
children: [
Stack(
children: [
CircleAvatar(
radius: 30,
backgroundImage: AssetImage('assets/images/template.png'),
),
if (isAdd)
Positioned(
right: 0,
bottom: 0,
child: CircleAvatar(
radius: 10,
backgroundColor: Colors.blue,
child: Icon(Icons.add, size: 15, color: Colors.white),
),
),
if (isActive)
Positioned(
right: 0,
bottom: 0,
child: Container(
width: 15,
height: 15,
decoration: BoxDecoration(
color: Colors.green,
shape: BoxShape.circle,
border: Border.all(color: Colors.white, width: 2),
),
),
),
],
),
SizedBox(height: 4),
Text(name, overflow: TextOverflow.ellipsis),
],
),
);
}
Widget _buildChatItem(String name, String message, String time, {bool isRead = false}) {
return ListTile(
leading: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
title: Text(name),
subtitle: Text(message),
trailing: Column(
mainAxisAlignment: MainAxisAlignment.center,
crossAxisAlignment: CrossAxisAlignment.end,
children: [
Text(time, style: TextStyle(fontSize: 12)),
if (isRead) Icon(Icons.check, size: 16, color: Colors.blue),
],
),
onTap: () {
Navigator.pushNamed(context, 'Intro');
},
);
}
Widget _buildAdItem() {
return ListTile(
leading: Image.asset('assets/images/template.png', width: 40, height: 40),
title: Row(
children: [
Text('Pixsellz'),
SizedBox(width: 4),
Container(
padding: EdgeInsets.symmetric(horizontal: 4, vertical: 2),
decoration: BoxDecoration(
color: Colors.grey[300],
borderRadius: BorderRadius.circular(4),
),
child: Text('Ad', style: TextStyle(fontSize: 10)),
),
],
),
subtitle: Text('Make design process easier...'),
trailing: Text('View More', style: TextStyle(color: Colors.blue)),
);
}
}